### PR TITLE
Add support for running controllers using kiam via clusterctl

### PR DESCRIPTION
**Pull request by randomvariable**
Signed-off-by: Naadir Jeewa <jeewan@vmware.com>

**What this PR does / why we need it**:
Adds annotations for running the controllers with IAM roles when either using kiam or EKS.

variable: AWS_CONTROLLER_IAM_ROLE

Sets `iam.amazonaws.com/role` and `eks.amazonaws.com/role-arn` annotations.

Going to create a follow up issue to add a variable reference / update quick start in CAPI repo.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Partial fix for #1964

### CODE REVIEW

**Comment by randomvariable** [comment]
/hold

/assign @michaelbeaumont @richardcase 

Do you know what the behaviour is on EKS if the role is blank?

**Comment by randomvariable** [comment]
/hold

@richardcase, @michaelbeaumont, do you know what the behaviour is on EKS if the role is blank?

**Comment by richardcase** [comment]
Good question, don't know off the top of my head....@michaelbeaumont do you?

I can try it out though if you want? Although i would hope it ignores a empty string. 

I've only ever used the `eks.amazonaws.com/role-arn` annotation with a service account.

(https://github.com/aws/amazon-eks-pod-identity-webhook)

**Comment by michaelbeaumont** [comment]
[It looks like it's ignored (for service accounts)](https://github.com/aws/amazon-eks-pod-identity-webhook/blob/2fd4becd434f92edab01cecee30ea4b12af93ab7/pkg/handler/handler.go#L225) but I also only know it as a service account annotation.

File: config/manager/manager_args_patch.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  namespace: system
+spec:
+  template:
+    spec:
+      containers:
+      - name: manager
+        args:
+        - "--metrics-addr=127.0.0.1:8080"
+        - "--enable-leader-election"
+        - "--feature-gates=EKS=${EXP_EKS:=false},EKSEnableIAM=${EXP_EKS_IAM:=false},EKSAllowAddRoles=${EXP_EKS_ADD_ROLES:=false}"
```
Comment (northrup) [inline config/manager/manager_args_patch.yaml:16]: Is this line needed in the non-EKS configuration given that the only evaluation happening is of EKS centric variables?

**Comment by northrup** [comment]
@randomvariable is there anything else needed to move this along?

**Comment by randomvariable** [comment]
@northrup testing mainly.

**Comment by cdenneen** [comment]
I think #2070 will be preferred over kiam?

**Comment by northrup** [comment]
@cdenneen possibly, the title maybe a misnomer but as it sits #2070 references EKS which I am not interested in running. I was looking for KIAM support for EC2 kubeadm bootstrapped nodes.

**Comment by cdenneen** [comment]
@northrup IRSA doesn’t require EKS and can work with kops or any other EC2 solution on AWS. Either way you are right as it probably should support both.

**Comment by randomvariable** [comment]
#2070 enables EKS clusters to consume IRSA
Actually, this PR is superseded by #2100 
so closing this one
/close

**Pull request closed without merging**